Unit1.pas

```diff
@@ -577,7 +577,12 @@ function MakeUni(s: ansistring): ansistring;
                     end;
                   end
                   else
-                    s := s + MakeUni(inttohex(Stack[stackb].value, 8));
+                  begin
```

Comment:
Mabe make an utility function that will return the display value

ex:
function GetDisplayValue(val: dword; size: byte): ansistring;
begin
if showdecimal then result := format('%.' + inttostr(size) + 'd', val)
else result := inttohex(val, size);
end;

then use it like this
s := s + MakeUni(GetDisplayValue(Stack[stackb].value, 8));

you can then use the same function on every place u need it